## 1. Data loading

In [416]:
import pandas as pd
import warnings

warnings.filterwarnings('ignore')

df = pd.read_csv("../Dataset/2021 New Coder Survey.csv")

df

Timestamp  \
0       7/1/2021 10:10:23   
1       7/1/2021 10:31:01   
2       7/1/2021 10:42:31   
3       7/1/2021 11:06:43   
4       7/1/2021 11:14:31   
...                   ...   
18121   9/29/2021 7:38:11   
18122  9/29/2021 12:52:59   
18123  9/29/2021 14:18:31   
18124   9/30/2021 4:04:32   
18125   10/1/2021 4:09:25   

      1. What is your biggest reason for learning to code?  \
0                           To succeed in current career     
1                                      To change careers     
2                                      To change careers     
3                                             As a hobby     
4                             To start your first career     
...                                                  ...     
18121                                  To change careers     
18122                                  To change careers     
18123                         To start your first career     
18124                                  To change careers     
18125                       To succeed in current career     

      2. What methods have you used to learn about coding? Please select all that apply.  \
0      Online resources, Books, In-person bootcamps, ...                                   
1              Online resources, Books, Online bootcamps                                   
2      Online resources, Books, Hackathons, Meetup.co...                                   
3                                Online resources, Books                                   
4              Online resources, Books, Online bootcamps                                   
...                                                  ...                                   
18121       Online resources, Books, In-person bootcamps                                   
18122              Online resources, In-person bootcamps                                   
18123                            Online resources, Books                                   
18124  Online resources, Books, In-person bootcamps, ...                                   
18125                            Online resources, Books                                   

      3. Which online learning resources have you found helpful? Please select all that apply.  \
0      freeCodeCamp, Mozilla Developer Network (MDN),...                                         
1      freeCodeCamp, Mozilla Developer Network (MDN),...                                         
2      freeCodeCamp, Mozilla Developer Network (MDN),...                                         
3      freeCodeCamp, Mozilla Developer Network (MDN),...                                         
4          freeCodeCamp, Stack Overflow, Coursera, Udemy                                         
...                                                  ...                                         
18121  Stack Overflow, EdX, Khan Academy, Udacity, Ci...                                         
18122  freeCodeCamp, Mozilla Developer Network (MDN),...                                         
18123  freeCodeCamp, Mozilla Developer Network (MDN),...                                         
18124  freeCodeCamp, Stack Overflow, Codecademy, Udem...                                         
18125  freeCodeCamp, Mozilla Developer Network (MDN),...                                         

      4. If you have attended in-person coding-related events before, which ones have you found helpful? Please select all that apply.  \
0              conferences, workshops, Meetup.com events                                                                                 
1      I haven't attended any in-person coding-relate...                                                                                 
2                                      Meetup.com events                                                                                 
3      I haven't attended any in-person coding-relate...                                

In [417]:
df.shape

(18126, 63)

from shape, we know there were **18126** participants in survey who answered 

63 - 1(timestamps) = **62 questions**.

we know from the original dataset description that there were 49 questions asked in survey: more columns imply some questions had subparts.

## 2. Exploratory Data Analysis (EDA)

In [418]:
df.head() # displaying top 5 rows

Timestamp 1. What is your biggest reason for learning to code?  \
0  7/1/2021 10:10:23                       To succeed in current career     
1  7/1/2021 10:31:01                                  To change careers     
2  7/1/2021 10:42:31                                  To change careers     
3  7/1/2021 11:06:43                                         As a hobby     
4  7/1/2021 11:14:31                         To start your first career     

  2. What methods have you used to learn about coding? Please select all that apply.  \
0  Online resources, Books, In-person bootcamps, ...                                   
1          Online resources, Books, Online bootcamps                                   
2  Online resources, Books, Hackathons, Meetup.co...                                   
3                            Online resources, Books                                   
4          Online resources, Books, Online bootcamps                                   

  3. Which online learning resources have you found helpful? Please select all that apply.  \
0  freeCodeCamp, Mozilla Developer Network (MDN),...                                         
1  freeCodeCamp, Mozilla Developer Network (MDN),...                                         
2  freeCodeCamp, Mozilla Developer Network (MDN),...                                         
3  freeCodeCamp, Mozilla Developer Network (MDN),...                                         
4      freeCodeCamp, Stack Overflow, Coursera, Udemy                                         

  4. If you have attended in-person coding-related events before, which ones have you found helpful? Please select all that apply.  \
0          conferences, workshops, Meetup.com events                                                                                 
1  I haven't attended any in-person coding-relate...                                                                                 
2                                  Meetup.com events                                                                                 
3  I haven't attended any in-person coding-relate...                                                                                 
4  I haven't attended any in-person coding-relate...                                                                                 

  5. If you have listened to coding-related podcasts before, which ones have you found helpful? Please select all that apply.  \
0                                      The Changelog                                                                            
1                 The Changelog, Code Newbie Podcast                                                                            
2                 I haven't listened to any podcasts                                                                            
3               Darknet Diaries, Real Python Podcast                                                                            
4                                  Talk Python to Me                                                                            

  6. If you have watched coding-related YouTube videos before, which channels have you found helpful? Please select all that apply.  \
0                              CS Dojo, freeCodeCamp                                                                                  
1  Adrian Twarog, Code with Ania Kubów, Coder Cod...                                                                                  
2  AmigosCode, Dev Ed, freeCodeCamp, Kevin Powell...                                                                                  
3                       freeCodeCamp, Traversy Media                                                                                  
4        freeCodeCamp, The Net Ninja, Traversy Media                                                                                  

   7. About how many hours do you spend learning each week?  \


In [419]:
df.tail() # displaying bottom 5 rows

Timestamp  \
18121   9/29/2021 7:38:11   
18122  9/29/2021 12:52:59   
18123  9/29/2021 14:18:31   
18124   9/30/2021 4:04:32   
18125   10/1/2021 4:09:25   

      1. What is your biggest reason for learning to code?  \
18121                                  To change careers     
18122                                  To change careers     
18123                         To start your first career     
18124                                  To change careers     
18125                       To succeed in current career     

      2. What methods have you used to learn about coding? Please select all that apply.  \
18121       Online resources, Books, In-person bootcamps                                   
18122              Online resources, In-person bootcamps                                   
18123                            Online resources, Books                                   
18124  Online resources, Books, In-person bootcamps, ...                                   
18125                            Online resources, Books                                   

      3. Which online learning resources have you found helpful? Please select all that apply.  \
18121  Stack Overflow, EdX, Khan Academy, Udacity, Ci...                                         
18122  freeCodeCamp, Mozilla Developer Network (MDN),...                                         
18123  freeCodeCamp, Mozilla Developer Network (MDN),...                                         
18124  freeCodeCamp, Stack Overflow, Codecademy, Udem...                                         
18125  freeCodeCamp, Mozilla Developer Network (MDN),...                                         

      4. If you have attended in-person coding-related events before, which ones have you found helpful? Please select all that apply.  \
18121                             conferences, workshops                                                                                 
18122  I haven't attended any in-person coding-relate...                                                                                 
18123  I haven't attended any in-person coding-relate...                                                                                 
18124  I haven't attended any in-person coding-relate...                                                                                 
18125  I haven't attended any in-person coding-relate...                                                                                 

      5. If you have listened to coding-related podcasts before, which ones have you found helpful? Please select all that apply.  \
18121                 Le code a changé (RadioFrance app)                                                                            
18122                 I haven't listened to any podcasts                                                                            
18123                 I haven't listened to any podcasts                                                                            
18124                 I haven't listened to any podcasts                                                                            
18125                 I haven't listened to any podcasts                                                                            

      6. If you have watched coding-related YouTube videos before, which channels have you found helpful? Please select all that apply.  \
18121  Hashtag Programação, Curso em Video, Boson Tre...                                                                                  
18122                                       freeCodeCamp                                                                                  
18123  Ben Awad, Coder Coder, Coding Train, Dev Ed, f...                                                                                  
18124  AmigosCode, Code with Ania Kubów, freeCodeCamp...                                                                                  
181

In [420]:
df.info() # displaying information like null count, datatype and name for each column

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 18126 entries, 0 to 18125
Data columns (total 63 columns):
 #   Column                                                                                                                                                                      Non-Null Count  Dtype  
---  ------                                                                                                                                                                      --------------  -----  
 0   Timestamp                                                                                                                                                                   18126 non-null  object 
 1   1. What is your biggest reason for learning to code?                                                                                                                        17991 non-null  object 
 2   2. What methods have you used to learn about coding? Please select all that apply.

from info() we note following:

**Numerical: 5 (float64)**

**Categorical: 58 (object)**

majority of the features are categorical and need to be changed according to what they represent, into numerical format for further usage.

In [421]:
print("Pandas Memory Usage:", df.memory_usage().sum(), 'Bytes') # memory usage while loading dataset

Pandas Memory Usage: 9135632 Bytes


In [422]:
print("Pandas Deep Memory Usage:", df.memory_usage(deep=True).sum(), 'Bytes') # deep memory usage while loading dataset, keeping each column into focus in terms of memory usage

Pandas Deep Memory Usage: 71443176 Bytes


**memory usage in loading data (pandas): 9135632 bytes : 8.9 MB**

**deep memory usage: 71443176 bytes : 69.8 MB**

In [423]:
df.describe(percentiles=[0.1,0.25,0.5,0.75,0.99]) # describes numerical interpretations of the dataset in terms of mean, max, min, quartiles, etc.

# percentiles considered: 10, 25, 50, 75, 99

7. About how many hours do you spend learning each week?  \
count                                       16603.000000          
mean                                           12.654852          
std                                            13.583577          
min                                             0.000000          
10%                                             1.500000          
25%                                             4.000000          
50%                                             8.000000          
75%                                            19.000000          
99%                                            64.000000          
max                                           150.000000          

       23. How old are you?  \
count          17103.000000   
mean              27.107420   
std               10.145221   
min                6.000000   
10%               17.000000   
25%               20.000000   
50%               25.000000   
75%               32.000000   
99%               60.000000   
max              120.000000   

       41. Before you got your last job, how many months did you spend looking for a job?  \
count                                       1.235200e+04                                    
mean                                        5.642575e+17                                    
std                                         6.271125e+19                                    
min                                         0.000000e+00                                    
10%                                         0.000000e+00                                    
25%                                         0.000000e+00                                    
50%                                         1.000000e+00                                    
75%                                         4.000000e+00                                    
99%                                         3.600000e+01                                    
max                                         6.969700e+21                                    

       42. If you are working, thinking about the next 12 months, how likely do you think it is that you will lose your job or be laid off?    \
count                                       10989.000000                                                                                        
mean                                            2.999272                                                                                        
std                                             2.539495                                                                                        
min                                             1.000000                                                                                        
10%                                             1.000000                                                                                        
25%                                             1.000000                                                                                        
50%                                             2.000000                                                                                        
75%                                             5.000000                                                                                        
99%                                            10.000000                                                                                        
max                                            10.000000                                                                                        

       43. If you are working, how easy would it be for you to find a job with another employer with approximately the same income and fringe benefits you now have?   
count                                       10914.000000                                                                                   

In [424]:
null_counts = df.isnull().sum().to_dict()

print("Column : null count\n")

for key,value in null_counts.items():
    print(key,":",value)

Column : null count

Timestamp : 0
1. What is your biggest reason for learning to code? : 135
2. What methods have you used to learn about coding? Please select all that apply. : 289
3. Which online learning resources have you found helpful? Please select all that apply. : 752
4. If you have attended in-person coding-related events before, which ones have you found helpful? Please select all that apply. : 1539
5. If you have listened to coding-related podcasts before, which ones have you found helpful? Please select all that apply. : 1536
6. If you have watched coding-related YouTube videos before, which channels have you found helpful? Please select all that apply. : 1257
7. About how many hours do you spend learning each week? : 1523
8. About how many months have you been programming? : 1431
9. Aside from university tuition, about how much money have you spent on learning to code so far (in US Dollars)? : 1973
10. Are you already employed in a software development job? : 453
11. If y

In [425]:
for i in df.columns: # printing column names to find complex questions with subparts
    print(i)

Timestamp
1. What is your biggest reason for learning to code?
2. What methods have you used to learn about coding? Please select all that apply.
3. Which online learning resources have you found helpful? Please select all that apply.
4. If you have attended in-person coding-related events before, which ones have you found helpful? Please select all that apply.
5. If you have listened to coding-related podcasts before, which ones have you found helpful? Please select all that apply.
6. If you have watched coding-related YouTube videos before, which channels have you found helpful? Please select all that apply.
7. About how many hours do you spend learning each week?
8. About how many months have you been programming?
9. Aside from university tuition, about how much money have you spent on learning to code so far (in US Dollars)?
10. Are you already employed in a software development job?
11. If you are already employed as a developer, is this your first software development job? 
12. I

Ques like:

39) How much debt does your household have?

45) Please tell us how satisfied you are with each of these following aspects of your present job

need to be dealt with as separable columns for max utilization.

---

Since majority of the data belongs to categorical type, we first feature engineer things to further process data.

---

## 3. Feature Engineering

In [426]:
column_names = ['Reason','Methods_Used_Learning','Helpful_Learning_Resources','Attended_Coding_Events','Listened_Coding_Podcasts','Watched_Coding_Videos','Hours_Spent_Learning_Weekly','Months_Programming','Money_Spent_Learning','Employed_SDE','First_SDE_Job','Not-SDE_Interested','SDE_Preference','interested Careers','Time_Application-SDE','Money-SDE_Expectations','Reasons-SDE','Ideal-Days_WFH-WFO_Weekly','Willing_Relocate','Employment_Status','Work-Field_if_Employed','Salary_Last_Year','Age','Self-Representation','Group-Identity','Part_of_World','State_if_US','City_Population','Citizenship_Country-diff-Live_Country','Ethnic_Minority_in_Live_Country','English_Second_Lang','Highest_Qualification','University-Field_of_Study','Current_Student_Enroll_Status','Marital Status','Children(B,S,A)','Financial_Dependents','No-Income_Survival_3-Months','Debt Car_Loans','Debt Credit_Cards','Debt Medical_Loans','Debt Mortgages','Debt Payday_Loans','Debt Student_Loans','Debt Other','Length_of_Employment','Last_Job-Hunt_Time_Spent','Chances_Of_Laid-off_12-Months','Time_Taken_Same-benefits_New_Employment','Under-Employed(Below_Education_Level)','Job-SatisFaction Earnings','Job-SatisFaction Benefits','Job-SatisFaction Job security','Job-SatisFaction Work-life balance','Job-SatisFaction Professional growth','Job-SatisFaction company culture','Job-SatisFaction Diversity','Job-SatisFaction Weekly workload','Time_Commute_Job','Previous_Military_Enlistment','Disability_Benefits-Government','High-Speed_Internet-Home']

We will be changing column names, which are currently questions, into representational names for ease-of-reference.

In [427]:
for i in range(1,len(column_names)+1):

    print("changing",list(df.columns)[i],"into:",column_names[i-1])

    df = df.rename(columns={list(df.columns)[i] : column_names[i-1]})

changing 1. What is your biggest reason for learning to code? into: Reason
changing 2. What methods have you used to learn about coding? Please select all that apply. into: Methods_Used_Learning
changing 3. Which online learning resources have you found helpful? Please select all that apply. into: Helpful_Learning_Resources
changing 4. If you have attended in-person coding-related events before, which ones have you found helpful? Please select all that apply. into: Attended_Coding_Events
changing 5. If you have listened to coding-related podcasts before, which ones have you found helpful? Please select all that apply. into: Listened_Coding_Podcasts
changing 6. If you have watched coding-related YouTube videos before, which channels have you found helpful? Please select all that apply. into: Watched_Coding_Videos
changing 7. About how many hours do you spend learning each week? into: Hours_Spent_Learning_Weekly
changing 8. About how many months have you been programming? into: Months_Pr

Feature names have been changed appropriately. Now we need to turn categorical data into either representational forms for further processing or Label Encode them.

In [429]:
df

Timestamp                        Reason  \
0       7/1/2021 10:10:23  To succeed in current career   
1       7/1/2021 10:31:01             To change careers   
2       7/1/2021 10:42:31             To change careers   
3       7/1/2021 11:06:43                    As a hobby   
4       7/1/2021 11:14:31    To start your first career   
...                   ...                           ...   
18121   9/29/2021 7:38:11             To change careers   
18122  9/29/2021 12:52:59             To change careers   
18123  9/29/2021 14:18:31    To start your first career   
18124   9/30/2021 4:04:32             To change careers   
18125   10/1/2021 4:09:25  To succeed in current career   

                                   Methods_Used_Learning  \
0      Online resources, Books, In-person bootcamps, ...   
1              Online resources, Books, Online bootcamps   
2      Online resources, Books, Hackathons, Meetup.co...   
3                                Online resources, Books   
4              Online resources, Books, Online bootcamps   
...                                                  ...   
18121       Online resources, Books, In-person bootcamps   
18122              Online resources, In-person bootcamps   
18123                            Online resources, Books   
18124  Online resources, Books, In-person bootcamps, ...   
18125                            Online resources, Books   

                              Helpful_Learning_Resources  \
0      freeCodeCamp, Mozilla Developer Network (MDN),...   
1      freeCodeCamp, Mozilla Developer Network (MDN),...   
2      freeCodeCamp, Mozilla Developer Network (MDN),...   
3      freeCodeCamp, Mozilla Developer Network (MDN),...   
4          freeCodeCamp, Stack Overflow, Coursera, Udemy   
...                                                  ...   
18121  Stack Overflow, EdX, Khan Academy, Udacity, Ci...   
18122  freeCodeCamp, Mozilla Developer Network (MDN),...   
18123  freeCodeCamp, Mozilla Developer Network (MDN),...   
18124  freeCodeCamp, Stack Overflow, Codecademy, Udem...   
18125  freeCodeCamp, Mozilla Developer Network (MDN),...   

                                  Attended_Coding_Events  \
0              conferences, workshops, Meetup.com events   
1      I haven't attended any in-person coding-relate...   
2                                      Meetup.com events   
3      I haven't attended any in-person coding-relate...   
4      I haven't attended any in-person coding-relate...   
...                                                  ...   
18121                             conferences, workshops   
18122  I haven't attended any in-person coding-relate...   
18123  I haven't attended any in-person coding-relate...   
18124  I haven't attended any in-person coding-relate...   
18125  I haven't attended any in-person coding-relate...   

                   Listened_Coding_Podcasts  \
0                             The Changelog   
1        The Changelog, Code Newbie Podcast   
2        I haven't listened to any podcasts   
3      Darknet Diaries, Real Python Podcast   
4                         Talk Python to Me   
...                                     ...   
18121    Le code a changé (RadioFrance app)   
18122    I haven't listened to any podcasts   
18123    I haven't listened to any podcasts   
18124    I haven't listened to any podcasts   
18125    I haven't listened to any podcasts   

                                   Watched_Coding_Videos  \
0                                  CS Dojo, freeCodeCamp   
1      Adrian Twarog, Code with Ania Kubów, Coder Cod...   
2      AmigosCode, Dev Ed, freeCodeCamp, Kevin Powell...   
3                           freeCodeCamp, Traversy Media   
4            freeCodeCamp, The Net Ninja, Traversy Media   
...                                                  ...   
18121  Hashtag Programação, Curso em Video, Boson Tre...   
18122                                       freeCodeCamp   
18123  Ben Awad, Coder Coder, Coding Tr

Now, we focus on our target: **Salary_Last_Year** and remove features inconsequential to this.

*The above is based on the fact that, subjective reasons and statements do not influence salary: for example Reason or the question: 1. What is your biggest reason for learning to code? does not affect salary earned due to the fact that this question is based on psychologial inclination towards preferred job.*

In [430]:
df1 = df.copy().drop(columns=['Timestamp','Reason','Reasons-SDE','interested Careers'])

changing salary to numerical

In [431]:
import numpy as np

for i in range(len(df1['Salary_Last_Year'])):
    value = df1['Salary_Last_Year'][i]
    if value in ["I don't want to answer", "I don’t know", np.nan]:
        df1.loc[i, 'Salary_Last_Year'] = np.random.choice([0, 1])
    elif value in ['$1,000 to $2,999', '$15,000 to $19,999', '$5,000 to $6,999', '$7,000 to $9,999', 'Under $1,000', '$3,000 to $4,999', '$10,000 to $14,999', '$20,000 to $24,999']:
        df1.loc[i, 'Salary_Last_Year'] = 0
    else:
        df1.loc[i, 'Salary_Last_Year'] = 1

changing disabilities benefit to numerical

In [432]:
df1['Disability_Benefits-Government'] = df1['Disability_Benefits-Government'].map({'Yes': 1, 'No': 0})

median_value = df1['Disability_Benefits-Government'].median()

df1['Disability_Benefits-Government'].fillna(median_value, inplace=True)

changing debts and combining them for total debt

In [433]:
debt_columns = ['Debt Car_Loans', 'Debt Credit_Cards', 'Debt Medical_Loans', 
                'Debt Mortgages', 'Debt Payday_Loans', 'Debt Student_Loans', 'Debt Other']

for j in debt_columns:
    for i in range(len(df1[j])):
        value = df1[j][i]
        if value == "<$10k":
            df1.loc[i, j] = 5000
        elif value == "$10k- $25k":
            df1.loc[i, j] = 15000
        elif value == "$26k- $50k":
            df1.loc[i, j] = 40000
        elif value == "$51k- $100k":
            df1.loc[i, j] = 80000
        elif value == "$101k- $200k":
            df1.loc[i, j] = 160000 
        elif value == "$201k- $500k":
            df1.loc[i, j] = 400000
        elif value == ">$500k":
            df1.loc[i, j] = 600000
        else:
            df1.loc[i, j] = 0

In [434]:
df1[debt_columns] = df1[debt_columns].astype(float)

df1['Total Debt'] = df1[debt_columns].sum(axis=1)

df1 = df1.drop(columns=debt_columns,axis=1)

changing high speed internet availability to numerical

In [435]:
df1['High-Speed_Internet-Home'] = df1['High-Speed_Internet-Home'].map({'Yes': 1, 'No': 0})

median_value = df1['High-Speed_Internet-Home'].median()

df1['High-Speed_Internet-Home'].fillna(median_value, inplace=True)

filling age null values with mean age value

In [436]:
mean_value = df1['Age'].mean()

df1['Age'].fillna(mean_value, inplace=True)

turning all job satisfaction columns into numerical format and fitting in median values

In [437]:
from sklearn.preprocessing import LabelEncoder

for i in df1.columns:
    encoder = LabelEncoder()
    if "Job-SatisFaction" in i:
        df1[i] = encoder.fit_transform(df1[i])
        median_value = df1[i].median()
        df1[i].fillna(median_value, inplace=True)

fitting in null values by turning money spent into float

In [439]:
df1['Money_Spent_Learning'] = df1['Money_Spent_Learning'].astype(float)

mean_value = df1['Money_Spent_Learning'].mean()

df1['Money_Spent_Learning'].fillna(mean_value, inplace=True)

turning into numerical and fitting in values for under-employed column

In [440]:
from sklearn.preprocessing import LabelEncoder

encoder = LabelEncoder()
df1['Under-Employed(Below_Education_Level)'] = encoder.fit_transform(df1['Under-Employed(Below_Education_Level)'])
median_value = df1['Under-Employed(Below_Education_Level)'].median()
df1['Under-Employed(Below_Education_Level)'].fillna(median_value, inplace=True)

fitting in null values for months spent in programming

In [441]:
df1['Months_Programming'] = df1['Months_Programming'].astype(float)

mean_value = df1['Months_Programming'].mean()

df1['Months_Programming'].fillna(0, inplace=True)

fitting in values and turning values for previous military enlistment

In [446]:
from sklearn.preprocessing import LabelEncoder

encoder = LabelEncoder()
df1['Previous_Military_Enlistment'] = encoder.fit_transform(df1['Previous_Military_Enlistment'])
median_value = df1['Previous_Military_Enlistment'].median()
df1['Previous_Military_Enlistment'].fillna(median_value, inplace=True)

fitting in values and turning values for chance of getting laid off in next 12 months

In [452]:
mean_value = df1['Chances_Of_Laid-off_12-Months'].mean()

df1['Chances_Of_Laid-off_12-Months'].fillna(0, inplace=True)

df1['Chances_Of_Laid-off_12-Months'] = df1['Chances_Of_Laid-off_12-Months'].astype(int)

fitting in values and turning values for sde employement

In [456]:
from sklearn.preprocessing import LabelEncoder

encoder = LabelEncoder()
df1['Employed_SDE'] = encoder.fit_transform(df1['Employed_SDE'])
median_value = df1['Employed_SDE'].median()
df1['Employed_SDE'].fillna(median_value, inplace=True)

fitting in values and turning values for willing to relocate

In [459]:
from sklearn.preprocessing import LabelEncoder

encoder = LabelEncoder()
df1['Willing_Relocate'] = encoder.fit_transform(df1['Willing_Relocate'])
median_value = df1['Willing_Relocate'].median()
df1['Willing_Relocate'].fillna(median_value, inplace=True)

fitting in values and turning values for city population

In [464]:
from sklearn.preprocessing import LabelEncoder

encoder = LabelEncoder()
df1['City_Population'] = encoder.fit_transform(df1['City_Population'])
median_value = df1['City_Population'].median()
df1['City_Population'].fillna(median_value, inplace=True)

fitting in values and turning values for financial dependents

In [469]:
from sklearn.preprocessing import LabelEncoder

encoder = LabelEncoder()
df1['Financial_Dependents'] = encoder.fit_transform(df1['Financial_Dependents'])
median_value = df1['Financial_Dependents'].median()
df1['Financial_Dependents'].fillna(median_value, inplace=True)

fitting in values and turning values for survival chances of no income in 3 months

In [472]:
from sklearn.preprocessing import LabelEncoder

encoder = LabelEncoder()
df1['No-Income_Survival_3-Months'] = encoder.fit_transform(df1['No-Income_Survival_3-Months'])
median_value = df1['No-Income_Survival_3-Months'].median()
df1['No-Income_Survival_3-Months'].fillna(median_value, inplace=True)

fitting in values and turning values for english being a second language

In [475]:
from sklearn.preprocessing import LabelEncoder

encoder = LabelEncoder()
df1['English_Second_Lang'] = encoder.fit_transform(df1['English_Second_Lang'])
median_value = df1['English_Second_Lang'].median()
df1['English_Second_Lang'].fillna(median_value, inplace=True)

fitting in values and turning values for children in family

In [479]:
from sklearn.preprocessing import LabelEncoder

encoder = LabelEncoder()
df1['Children(B,S,A)'] = encoder.fit_transform(df1['Children(B,S,A)'])
median_value = df1['Children(B,S,A)'].median()
df1['Children(B,S,A)'].fillna(median_value, inplace=True)

fitting in values and turning values for highest qualification

In [482]:
from sklearn.preprocessing import LabelEncoder

encoder = LabelEncoder()
df1['Highest_Qualification'] = encoder.fit_transform(df1['Highest_Qualification'])
median_value = df1['Highest_Qualification'].median()
df1['Highest_Qualification'].fillna(median_value, inplace=True)

fitting in values and turning values for marital status

In [486]:
from sklearn.preprocessing import LabelEncoder

encoder = LabelEncoder()
df1['Marital Status'] = encoder.fit_transform(df1['Marital Status'])
median_value = df1['Marital Status'].median()
df1['Marital Status'].fillna(median_value, inplace=True)

fitting in values and turning values for length of employment

In [489]:
from sklearn.preprocessing import LabelEncoder

encoder = LabelEncoder()
df1['Length_of_Employment'] = encoder.fit_transform(df1['Length_of_Employment'])
median_value = df1['Length_of_Employment'].median()
df1['Length_of_Employment'].fillna(median_value, inplace=True)

fitting in values and turning values for time spent in job hunting

In [491]:
df1['Last_Job-Hunt_Time_Spent'].fillna(0, inplace=True)

for i in range(len(df1['Last_Job-Hunt_Time_Spent'])):
    if i > 100:
        df1['Last_Job-Hunt_Time_Spent'][i] = 0

fitting in values and turning values for ethinical minority in country

In [494]:
from sklearn.preprocessing import LabelEncoder

encoder = LabelEncoder()
df1['Ethnic_Minority_in_Live_Country'] = encoder.fit_transform(df1['Ethnic_Minority_in_Live_Country'])
median_value = df1['Ethnic_Minority_in_Live_Country'].median()
df1['Ethnic_Minority_in_Live_Country'].fillna(median_value, inplace=True)

fitting in values and turning values for values if the country of citizen is different than country of residence.

In [497]:
from sklearn.preprocessing import LabelEncoder

encoder = LabelEncoder()
df1['Citizenship_Country-diff-Live_Country'] = encoder.fit_transform(df1['Citizenship_Country-diff-Live_Country'])
median_value = df1['Citizenship_Country-diff-Live_Country'].median()
df1['Citizenship_Country-diff-Live_Country'].fillna(median_value, inplace=True)

fitting in values and turning values for states in US

In [501]:
from sklearn.preprocessing import LabelEncoder

encoder = LabelEncoder()
df1['State_if_US'] = encoder.fit_transform(df1['State_if_US'])
df1['State_if_US'].fillna(0, inplace=True)

fitting in values and turning values for current Student enrollment status

In [504]:
from sklearn.preprocessing import LabelEncoder

encoder = LabelEncoder()
df1['Current_Student_Enroll_Status'] = encoder.fit_transform(df1['Current_Student_Enroll_Status'])
median_value = df1['Current_Student_Enroll_Status'].median()
df1['Current_Student_Enroll_Status'].fillna(median_value, inplace=True)

fitting in values and turning values for Employement status

In [506]:
from sklearn.preprocessing import LabelEncoder

encoder = LabelEncoder()
df1['Employment_Status'] = encoder.fit_transform(df1['Employment_Status'])
median_value = df1['Employment_Status'].median()
df1['Employment_Status'].fillna(median_value, inplace=True)